daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1578,47 +1570,79 @@ object Main extends Logging {
     1
   }
 
+  object ExitCode extends Enumeration {
+
+    val Success = Value(0)
+    val Failure = Value(1)
+
+    val FileNotFound = Value(2)
+    val OutOfMemory = Value(3)
+    val BugFound = Value(4)
+    val NotYetImplemented = Value(5)
+
+
+    val ParseError = Value(20)
+    val UnparseError = Value(21)
+    val GenerateCodeError = Value(23)
+    val TestError = Value(24)
+    val PerformanceTestError = Value(25)
+
+
+    val LeftOverData = Value(31)
+    val InvalidParserException = Value(32)
+    val BadExternalVariable = Value(33)
+    val UserDefinedFunctionError = Value(34)
+    val UnableToCreateProcessor = Value(35)
+
+    val Usage = Value(64)
+
+  }
+
   def main(arguments: Array[String]): Unit = {
+
     val ret = try {
       run(arguments)
     } catch {
       case s: scala.util.control.ControlThrowable => throw s
       case e: java.io.FileNotFoundException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
-      }
-      case e: InvalidParserException => {
-        log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.FileNotFound
       }
       case e: ExternalVariableException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.BadExternalVariable
       }
       case e: BindingException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.BadExternalVariable
       }
       case e: NotYetImplementedException => {
         nyiFound(e)
+        ExitCode.NotYetImplemented
       }
       case e: TDMLException => {
         log(LogLevel.Error, "%s", e.getMessage())
-        1
+        ExitCode.TestError
       }
       case e: OutOfMemoryError => {
         oomError(e)
+        ExitCode.OutOfMemory
       }
       case e: UserDefinedFunctionFatalErrorException => {
         log(LogLevel.Error, "%s", e.getMessage())
         e.printStackTrace()
-        1
+        ExitCode.UserDefinedFunctionError
+      }
+      case e: DebuggerExitException => {
+        log(LogLevel.Error, "%s", e.getMessage())
+        ExitCode.Failure
```

Comment:
Should be ExitCode.Success. A quit command is a normal exit.

Reply:
I could see an argument for it being both. It is definitely a normal exit, but it also means we exited without generating a complete infoset. I could imagine someone using a script that triggers the CLI debugger, but still wants the script to be able to detect if the infoset was complete or not once the debugger finishes.

Reply:
Also, we should remove this `log()` call. Although this is an exception, it's an expected exception when the user quits the debugger. When they do this, they'll know that things are going to exit, so they don't need a message confirming things.